components/calendar/src/any_calendar.rs

```diff
@@ -611,6 +708,16 @@ impl AnyCalendarKind {
             .and_then(Self::from_bcp47)
     }
 
+    /// Set the `u-ca` extension on a DataLocale to the calendar represented
+    /// by this type
+    pub fn set_on_data_locale(&self, l: &mut DataLocale) {
```

Comment:
Nit (optional): I'm not really a fan of this function. The DataLocale API is fine and more explicit.

The only thing this really accomplishes is hiding the key!(ca). Maybe make that an exported const in this crate?

Reply:
It's called thrice and it encapsulates that logic nicely; this way other code doesn't actually have to muck with AnyCalendarKind too much